controllers/taskRequestsv2.ts

```diff
@@ -0,0 +1,174 @@
+import { REQUEST_STATE, TASK_REQUEST_MESSAGES } from "../constants/requests";
+import { TASK_REQUEST_TYPE } from "../constants/taskRequests";
+import { addLog } from "../models/logs";
+import { createRequest, getRequestByKeyValues } from "../models/requests";
+import { fetchTask } from "../models/tasks";
+import { fetchUser } from "../models/users";
+import { fetchIssuesById } from "../services/githubService";
+import { CustomResponse } from "../typeDefinitions/global";
+import { userData } from "../types/global";
+import { TaskRequestRequest } from "../types/taskRequests";
+
+export const createTaskRequestController = async (req: TaskRequestRequest, res: CustomResponse) => {
+  const taskRequestData = req.body;
+  const requestedBy = req?.userData?.id;
+
+  if (!requestedBy) {
+    return res.boom.unauthorized();
+  }
+
+  if (req.userData.id !== taskRequestData.userId && !req.userData.roles?.super_user) {
+    return res.boom.forbidden(TASK_REQUEST_MESSAGES.NOT_AUTHORIZED_TO_CREATE_REQUEST);
+  }
```

Comment:
Please ensure that user inputs are sanitised to prevent injection attacks. Also, Can you please verify that sensitive data is handled securely.

Reply:
Which type of injection attack?

Reply:
And which sensitive data is not handled securely?

Reply:
I just said to double-check that the sensitive data is handled securely. rest the PR is looking fine from my side.

Reply:
Can you please approve this PR if everything looks good to you?